### ISSUE TITLE
Ability to use Webhook response in form ? 

### ISSUE THREAD

**Issue by imadbakir** [formio/formio#698]
Hello, 

I'm doing some experiments with webhooks, in a case in witch I need to deal with the request response afterwards (example: set form fields values from response object). 
what I have noticed is that I can assign the response to ExternalIds, and also I can see the data in "metadata" on the submission object, but I have no "tool" to access metadata, or externalIds (for example in calculated/default values), 
is there a way we can achieve that? 
or maybe execute code on "response" ? 

Thanks!

**Comment by randallknutson** [comment]
metadata is not available in calculated fields. We could add this but there may be a better way of facilitating what you are looking for. You may be able to use blocking webhooks to affect the submission value and response that is then passed back to the client. This would allow you to add additional data, such as the submission state and then respond on the front end.

If you don't use webhook blocking, it won't wait for a response from the webhook before returning the result to the client.

**Comment by imadbakir** [comment]
> You may be able to use blocking webhooks to affect the submission value and response that is then passed back to the client. This would allow you to add additional data, such as the submission state and then respond on the front end.
> 
> 

Thanks for the reply, 
Do you by blocking webhooks mean the "Wait for webhook response before continuing actions" option? 
if so, how am I supposed to use that to achieve my wanted behavior (fill the form from Get Request data)? 
@randallknutson

**Comment by randallknutson** [comment]
It looks like there currently is no other way to modify the results of the submission except the added data to metadata during the webhook cycle. There are a couple ways we could solution this for you.

1. You can add some code to your application that listens for the load submission event and move the data from metadata to data. This would then be available in the form.
2. We could add metadata to the available variables when calculating values/logic/etc. 

If you wanted to do 2 in a pull request, the code would be somewhere around https://github.com/formio/formio.js/blob/master/src/Component.js#L497-L505 or https://github.com/formio/formio.js/blob/master/src/components/base/Base.js#L2155-L2158

**Comment by imadbakir** [comment]
@randallknutson thanks a lot, looking into the 2nd option then!

**Comment by randallknutson** [comment]
Yes. We do builds of these and it would be available on our platform in a matter of days/week or so depending on the deployment process.

**Comment by nowickimt** [comment]
Looking forward to seeing what you guys come up with - I have a very similar need to call a webhook, wait for a response, then display data related to that response BEFORE the final submission happens.

**Comment by randallknutson** [comment]
@nowickimt Here is an example of how to do what you are asking for:
http://jsfiddle.net/randallknutson/eLp1hrm2/

**Comment by edwinanciani** [comment]
Hi @nowickimt, these are two examples that you could use to validate or fill the submission data in the form before the final submission happens. [Custom Validation](http://jsfiddle.net/randallknutson/eLp1hrm2/) and [External Data Load](http://formio.github.io/formio.js/app/examples/externalload.html)
